File: benchmark/closure_loop.py
def make_accumulator(seed):
    total = seed
    def add(n):
        nonlocal total
        total = total + n
        return total
    return add

add = make_accumulator(0)
check = 0
for i in range(5000000):
    check = add(i)
print(check)

File: benchmark/closure_loop.lua
local function make_accumulator(seed)
    local total = seed
    return function(n)
        total = total + n
        return total
    end
end

local add = make_accumulator(0)
local check = 0
for i = 0, 4999999 do
    check = add(i)
end
print(check)
